timestamp-api/src/main/java/com/palantir/timestamp/AbstractTimestampBoundStoreWithId.java

```diff
@@ -0,0 +1,203 @@
+                + " does not match this service's ID: {}. This may indicate that another timestamp service is"
+                + " running against this cassandra keyspace.";
+        String formattedMessage = MessageFormatter
+                .arrayFormat(message, new String[]{entryInDb.getIdAsString(), id.toString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
+
+    protected void throwStoringMultipleRunningTimestampServiceError(
+            TimestampBoundStoreEntry entryInDb,
+            TimestampBoundStoreEntry desiredNewEntry) {
+        String message = "Unable to CAS from {} to {}. Timestamp limit changed underneath us or another timestamp"
+                + " service ID detected. Limit in memory: {}, this service's ID: {}. Limit stored in DB: {},"
+                + " the ID stored in DB: {}. This may indicate that another timestamp service is running against"
+                + " this cassandra keyspace. This is likely caused by multiple copies of a service running without"
+                + " a configured set of leaders or a CLI being run with an embedded timestamp service against"
+                + " an already running service.";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit),
+                desiredNewEntry.getTimestampAsString(),
+                Long.toString(currentLimit),
+                id.toString(),
+                entryInDb.getTimestampAsString(),
+                entryInDb.getIdAsString()})
+                .getMessage();
+        logMessage(formattedMessage);
+        throw new MultipleRunningTimestampServiceError(formattedMessage);
+    }
+
+    protected void throwNewTimestampTooSmallException(TimestampBoundStoreEntry entryInDb) {
+        String message = "Cannot set cached timestamp bound value from {} to {}. The bounds must be increasing!";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                Long.toString(currentLimit), entryInDb.getTimestampAsString()}).getMessage();
+        logMessage(formattedMessage);
+        throw new IllegalArgumentException(formattedMessage);
+    }
+
+    public static void logUpdateUncheckedException(TimestampBoundStoreEntry entryInDb,
+            TimestampBoundStoreEntry desiredNewEntry) {
+        String message = "[CAS] Error trying to set from {} to {}";
+        String formattedMessage = MessageFormatter.arrayFormat(message, new String[]{
+                entryInDb.getTimestampAsString(), desiredNewEntry.getTimestampAsString()}).getMessage();
+        logMessage(formattedMessage);
+    }
+
+    private static void logMessage(String formattedMessage) {
+        log.error("Error: {}", formattedMessage);
+        log.error("Thread dump: {}", ThreadDumps.programmaticThreadDump());
```

Comment:
Why are these two separate log messages? Why not just add newline in message pattern and include in first log.error?